android_app/app/src/main/java/com/health/openscale/core/bluetooth/BluetoothStandardWeightProfile.java

```diff
@@ -19,85 +19,185 @@
                 break;
-            case 1:
-                // Write the current time
-                BluetoothBytesParser parser = new BluetoothBytesParser();
-                parser.setCurrentTime(Calendar.getInstance());
-                writeBytes(BluetoothGattUuid.SERVICE_CURRENT_TIME, BluetoothGattUuid.CHARACTERISTIC_CURRENT_TIME, parser.getValue());
+            case WRITE_CURRENT_TIME:
+                writeCurrentTime();
                 break;
-            case 2:
+            case SET_NOTIFY_WEIGHT_MEASUREMENT:
                 // Turn on notification for Weight Service
                 setNotificationOn(BluetoothGattUuid.SERVICE_WEIGHT_SCALE, BluetoothGattUuid.CHARACTERISTIC_WEIGHT_MEASUREMENT);
                 break;
-            case 3:
+            case SET_NOTIFY_BODY_COMPOSITION_MEASUREMENT:
                 // Turn on notification for Body Composition Service
                 setNotificationOn(BluetoothGattUuid.SERVICE_BODY_COMPOSITION, BluetoothGattUuid.CHARACTERISTIC_BODY_COMPOSITION_MEASUREMENT);
                 break;
-            case 4:
-                // Turn on notification for User Data Service
+            case SET_NOTIFY_CHANGE_INCREMENT:
+                // Turn on notification for User Data Service Change Increment
                 setNotificationOn(BluetoothGattUuid.SERVICE_USER_DATA, BluetoothGattUuid.CHARACTERISTIC_CHANGE_INCREMENT);
-                setNotificationOn(BluetoothGattUuid.SERVICE_USER_DATA, BluetoothGattUuid.CHARACTERISTIC_USER_CONTROL_POINT);
                 break;
-            case 5:
+            case SET_INDICATION_USER_CONTROL_POINT:
+                // Turn on notification for User Control Point
+                setIndicationOn(BluetoothGattUuid.SERVICE_USER_DATA, BluetoothGattUuid.CHARACTERISTIC_USER_CONTROL_POINT);
+                break;
+            case SET_NOTIFY_BATTERY_LEVEL:
                 // Turn on notifications for Battery Service
-                setNotificationOn(BluetoothGattUuid.SERVICE_BATTERY_LEVEL, BluetoothGattUuid.CHARACTERISTIC_BATTERY_LEVEL);
+                if (setNotificationOn(BluetoothGattUuid.SERVICE_BATTERY_LEVEL, BluetoothGattUuid.CHARACTERISTIC_BATTERY_LEVEL)) {
+                    haveBatteryService = true;
+                }
+                else {
+                    haveBatteryService = false;
+                }
+                break;
+            case READ_BATTERY_LEVEL:
+                // read Battery Service
+                if (haveBatteryService) {
+                    readBytes(BluetoothGattUuid.SERVICE_BATTERY_LEVEL, BluetoothGattUuid.CHARACTERISTIC_BATTERY_LEVEL);
+                }
+                break;
+            case SET_NOTIFY_VENDOR_SPECIFIC_USER_LIST:
+                setNotifyVendorSpecificUserList();
+                break;
+            case REQUEST_VENDOR_SPECIFIC_USER_LIST:
+                scaleUserList.clear();
+                requestVendorSpecificUserList();
```

Comment:
We need a to stop the machine here:
```suggestion
                requestVendorSpecificUserList();
                stopMachineState();
                break;
```